### DISCUSSION TITLE
Tile Layer - Show previous tile until new tile is loaded and rendered

### DISCUSSION THREAD

**Discussion by abiswas97** [visgl/deck.gl#8078] [Q&A]
I'm trying to load tiled geoJSON data into a TileLayer, and am having difficulty with the 'stutter' when the previous layer disappears before the layer at the next zoom is rendered. 

It looks like the `refinementStrategy` takes care of this until data is loaded - but I think my bottleneck is when the shaders are being generated before the render of the fetched tile. 

Here is my setup
```
const layer = new TileLayer({
    id: id,
    minZoom: -metadata?.maxZoom ?? -7,
    maxZoom: 0,
    maxCacheSize: 500,
    maxRequests: 10,
    extent: metadata?.bounds,
    tileSize: 512,
    refinementStrategy: "best-available",
    coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
    getTileData: async (props) => {
      const { x, y, z } = props.index;
      const { signal } = props;

      if (signal?.aborted) {
        return null;
      }

      const loadData = async (url: string) => {
        const gzip = new GZipCompression();

        const dataFile = await fetchFile(`${url}/${z}/${x}_${y}.geojson.gz`);

        const data = gzip.decompress(await dataFile.arrayBuffer());
        return parse(data, GeoJSONLoader, { worker: true });
      };

      return loadData(url);
    },
    renderSubLayers: (props: any) => {
      return new GeoJsonLayer({
        id: `${id}-${props.tile.id}-geojson`,
        data: props.data,
        pickable: true,
        stroked: true,
        filled: true,
        lineWidthMinPixels: 1,
        getLineColor: (d: any): RGBAColor => d.properties.classification.color,
        getFillColor: (): RGBAColor => [0, 0, 0, 0],
        getLineWidth: 5,
      });
    },
  });
  ```
  
  What would be a potential strategy to have the previous layer stay visible until the current tiles are rendered?

**Comment by ibgreen** [comment]
Shaders should only be created once, they are cached and shared between layers specifically because shader compilation (or rather shader linking) tends to be so slow. 

So, if you have continuous stuttering it is probably not that. 

Which is not to say that there couldn't be some delay in the layers...

**Reply by abiswas97** [reply]
Once the layers are loaded, there is no more stuttering! But the initial switch from say z=1 to z=2 causes a stutter
